card/lib/card/view_name.rb

```diff
@@ -0,0 +1,44 @@
+# -*- encoding : utf-8 -*-
+require_dependency 'card/env'
+
+require 'smart_name'
+
+class Card
+  class ViewName < SmartName
+    @@name2viewnameobject = {}
+
+    class << self
+      def new obj
+        return obj if self.class===obj
+        str = Array===obj ? obj*joint : obj.to_s
+        if known_name = @@name2viewnameobject[str]
+          known_name
+        else
+          super str.strip
+        end
+      end
+    end
+
+    def initialize str
+      @s = str.to_s.strip
+      @s = @s.encode('UTF-8') if RUBYENCODING
+      @key = if @s.index(self.class.joint)
+          @parts = @s.split(/\s*#{JOINT_RE}\s*/)
+          @parts << '' if @s[-1,1] == self.class.joint
+          @simple = false
+          @parts.map { |p| p.to_name.key } * self.class.joint
+        else
+          @parts = [str]
+          @simple = true
+          str.empty? ? '' : simple_key
+        end
+      @@name2viewnameobject[str] = self
+    end
+
+    def simple_key
+      decoded.underscore.gsub(/[^#{OK4KEY_RE}]+/,'_').split(/_+/).reject(&:empty?)*'_'
+    end
```

Comment:
I think we are saying more or less the same thing.  I don't see any need for variants in code, and my point was that if you want the plural for the codename, you can have that, but only that codename (pluralized).  You could actually create two cards with codenames (i.e. edit_rule and edit_rules could be codenames of distinct cards, although that is probably a bad idea to do) that would match as names.

I thought that's why we were using name processing for view names (to get the code/user context translation on view names)?

Can't we implement a migration to convert those old views?  Presumably, these need to be supported as user space names (i.e. in inclusion/nest syntax), and not as codenames?